app/models/miq_ae_domain.rb

```diff
@@ -8,25 +10,54 @@ class MiqAeDomain < MiqAeNamespace
   EDITABLE_PROPERTIES_FOR_REMOTES = [:priority, :enabled].freeze
   AUTH_KEYS = %w[userid password].freeze
 
-  default_scope { roots.where(arel_table[:name].not_eq("$")) }
+  default_scope { roots.where.not(:name => "$") }
   validates :ancestry, :inclusion => {:in => [nil], :message => 'should be nil for Domain'}
 
   validates_presence_of :tenant, :message => "object is needed to own the domain"
   after_destroy :squeeze_priorities
+  after_destroy :clear_domain_cache
+  after_save :clear_domain_cache
   default_value_for :source,  USER_SOURCE
   default_value_for :enabled, false
   before_save :default_priority
   belongs_to :tenant
   belongs_to :git_repository, :dependent => :destroy
   validates_inclusion_of :source, :in => VALID_SOURCES
 
+  alias_attribute :domain_name, :name
+
   EXPORT_EXCLUDE_KEYS = [/^id$/, /^(?!tenant).*_id$/, /^created_on/, /^updated_on/,
                          /^updated_by/, /^reserved$/, /^commit_message/,
                          /^commit_time/, /^commit_sha/, /^ref$/, /^ref_type$/,
                          /^last_import_on/].freeze
 
   include TenancyMixin
 
+  def self.name_to_id(name)
+    return if name.nil?
+
+    load_domain_cache
+    name = name.downcase
+    @domain_names[name] ||= MiqAeDomain.unscoped.roots.where(:lower_name => name).pluck(:id).first
+  end
+
+  def self.id_to_name(domain_id)
+    return if domain_id.nil?
+
+    load_domain_cache
+    @domain_ids[domain_id] ||= MiqAeDomain.unscoped.roots.where(:id => domain_id).pluck(:name).first
+  end
+
+  delegate :clear_domain_cache, :to => self
+  def self.clear_domain_cache
+    @domain_ids = @domain_names = nil
+  end
+
+  def self.load_domain_cache
+    @domain_ids ||= Hash[*MiqAeDomain.unscoped.roots.pluck(:id, :name).flatten]
```

Comment:
Maybe make `MiqAeDomain.unscoped.roots` into a `scope`?

Reply:
Not sure how to get that to work.

I want a hash for `{"#{id1}" => #{"name1}", "#{id2}" => #{"name2}"}`

that suggestion gives me `[{:id =>"#{id1}", :name => "#{name1}"}, {:id =>"#{id2}", :name => "#{name2}"}]`

your suggestion produced an array of `[id, name]`

Reply:
Ah, I guess I miss read your code, my bad.

Yeah, then your code probably is the best option available then.  Though, probably just doing:

```ruby
MiqAeDomain.unscoped.roots.pluck(:id, :name).to_h
```

Would also achieve the same thing.

Also, just for clarification:

> I want a hash for {"#{id1}" => #{"name1}", "#{id2}" => #{"name2}"}

What you have and even what I suggested above will **not** have `:id`s as strings, but as ints.  Your example suggests otherwise.